protocol-impl/src/main/java/io/camunda/zeebe/protocol/impl/record/value/processinstance/ProcessInstanceModificationVariableInstruction.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.protocol.impl.record.value.processinstance;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import io.camunda.zeebe.msgpack.property.DocumentProperty;
+import io.camunda.zeebe.msgpack.property.StringProperty;
+import io.camunda.zeebe.msgpack.value.ObjectValue;
+import io.camunda.zeebe.protocol.impl.encoding.MsgPackConverter;
+import io.camunda.zeebe.protocol.record.value.ProcessInstanceModificationRecordValue.ProcessInstanceModificationVariableInstructionValue;
+import io.camunda.zeebe.util.buffer.BufferUtil;
+import java.util.Map;
+import org.agrona.DirectBuffer;
+
+@JsonIgnoreProperties({
+  /* 'encodedLength' is a technical field needed for MsgPack and inherited from ObjectValue; it has
+  no purpose in exported JSON records*/
+  "encodedLength"
+})
+public final class ProcessInstanceModificationVariableInstruction extends ObjectValue
+    implements ProcessInstanceModificationVariableInstructionValue {
+
+  private final DocumentProperty variablesProp = new DocumentProperty("variables");
+  private final StringProperty elementIdProp = new StringProperty("elementId", "");
```

Comment:
```suggestion
  private final StringProperty elementIdProp = new StringProperty("elementId");
```
 🔧

Reply:
I don't think this would work. The `elementId` of variable instructions is supposed to be optional, if it's not set then the variables are "globally" scoped for the process instance.

Reply:
I believe it will. Looking at `StringProperty` you can see it has 2 constructors:
```java
  public StringProperty(final String key) {
    super(key, new StringValue());
  }

  public StringProperty(final String key, final String defaultValue) {
    super(key, new StringValue(), new StringValue(defaultValue));
  }
```

So regardless it will always set a `StringValue`. In your case it passes an empty string. In the empty constructor a the `StringValue` does the exact same thing 😄 

```java
  public StringValue() {
    this(EMPTY_STRING);
  }
```

Reply:
TBH, I don't understand why it's not implemented as:
```java
public StringProperty(final String key) {
  this(key, "");
}
```

This would set the default string to `""` instead of `null`. I think we would always favor this, but I'm not sure 🤷

Reply:
Oooh never mind. So I was missing something obvious 😄 Time for weekend

Reply:
I don't think it's obvious actually 😆 Have a good weekend